services/web-app/components/catalog-sort/catalog-sort.js

```diff
@@ -6,65 +6,95 @@
  */
 import { Button, Column, Dropdown, Grid, Layer, Theme } from '@carbon/react'
 import { Grid as GridIcon, List as ListIcon } from '@carbon/react/icons'
+import clsx from 'clsx'
+import { useCallback, useEffect, useRef, useState } from 'react'
 
-import { mediaQueries, useMatchMedia } from '@/utils/media-query'
+import useEventListener from '@/utils/use-event-listener'
+import { mediaQueries, useMatchMedia } from '@/utils/use-match-media'
 
 import styles from './catalog-sort.module.scss'
 
 const CatalogSort = ({ onSort, onView, sort, view }) => {
   const isMd = useMatchMedia(mediaQueries.md)
+  const containerRef = useRef(null)
+  const [isSticky, setIsSticky] = useState(false)
 
-  const sortOptions = ['A–Z', 'Newest', 'Most used', 'Status']
+  const sortItems = [
+    {
+      id: 'a-z',
+      text: 'A–Z'
+    },
+    {
+      id: 'status',
+      text: 'Status'
+    }
+  ]
+
+  // Conditionally add a drop shadow through JavaScript because `position:sticky` doesn't support a
+  // `::stuck` pseudo-class to trigger the drop shadow. Header (48) + spacer (16) + search (48) =
+  // 112.
+  const scrollHandler = useCallback(() => {
+    if (containerRef.current && typeof window !== 'undefined') {
+      setIsSticky(containerRef.current.getBoundingClientRect().top === 112)
+    }
+  }, [])
+
+  useEventListener('scroll', scrollHandler)
+  useEffect(() => {
+    scrollHandler()
+  }, [scrollHandler])
```

Comment:
Depending on what use cases you envision for `useEventListener`, you could potentially put the `useCallback` inside of the `useEventListener` hook to remove the need for each hook user to memoize their callback before providing it as an arg.

Reply:
That gives an eslint warning in the static code analysis. It cool if I keep it as is?

![image](https://user-images.githubusercontent.com/1691245/145117399-2d294953-7a94-4ee9-b5fc-08da4273c24f.png)

Reply:
Ohhh. Sorry. I didn't realize it was getting used below in the `useEffect` hook as well.

Reply:
Ya! Just to make sure the handler fires once on page load too.